tests/unit/test_charm.py

```diff
@@ -285,7 +298,9 @@ def test_new_password(self):
         self.assertIsNotNone(re.fullmatch("[a-zA-Z0-9\b]{16}$", second_password))
         self.assertNotEqual(second_password, first_password)
 
-    def test_get_postgres_password(self):
+    @patch("charm.PostgresqlOperatorCharm._patch_pod_labels")
+    @patch("charm.PostgresqlOperatorCharm._create_resources")
+    def test_get_postgres_password(self, _, __):
         # Test for a None password.
         self.harness.add_relation(self._peer_relation, self.charm.app.name)
         self.assertIsNone(self.charm._get_postgres_password())
```

Comment:
Missing tests for these scenarios (i dont believe these are tested in the integration tests which is why I have this comment here, but please correct me if I am wrong on any of these)  : 

[this scenario](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L79-L82) 
[this](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L90-L93) 
[this](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L99-L101)
[this](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L107-L111)
[this](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L140-L141)
[this](https://github.com/canonical/postgresql-k8s-operator/blob/3b9be85fd040d2980e2b571d82039a1636ecb37e/src/charm.py#L154-L157)

After looking at the code, there other scenarios that are not tested in the unit tests for patroni.py + charm.py (I was lazy and didn't want to write them out)

But I am happy to write them all out if you'd like, or review your next commit, let me know what you prefer

Reply:
Synced offline about implementing them in another PR to not this PR bigger. Ticket DPE-399.